### PR TITLE
AEROGEAR-2575 Removal of the associating OpenShift build objects

### PR DESCRIPTION
**Pull request by vchepeli**
**JIRA** [AEROGEAR-2575](https://issues.jboss.org/browse/AEROGEAR-2575)
**CHANGELOG**
- [ ] delete jenkins builds
- [ ] delete jenkins build configs 
- [ ] delete jenkins secrets

**VERIFICATION**
1. Provision Mobile CI|CD
- [ ] `oc create -f file1.yml -n test1`
- [ ] `oc label bc/helloworld-android-pipeline1 mobile-client-build=true -n test1`
- [ ] `oc start-build helloworld-android-pipeline1 -n test1`
- [ ] `oc create secret generic helloworld-android-cert1 -n test1`
- [ ] `oc label secret/helloworld-android-cert1 mobile-client-build=true -n test1`
2. Deprovision Mobile CI|CD
- [ ] jenkins builds deleted
- [ ] jenkins build configs deleted
- [ ] jenkins build secrets deleted

```yaml
---
kind: BuildConfig
apiVersion: v1
metadata:
  name: helloworld-android-pipeline1
spec:
  source:
    git:
      uri: https://github.com/odra/helloworld-android-gradle.git
      ref: jenkins-apk-path
  strategy:
    jenkinsPipelineStrategy:
      env:
        -
          name: FH_CONFIG_CONTENT
          value: empty_content
        -
          name: BUILD_CONFIG
          value: debug
      jenkinsfilePath: Jenkinsfile
```

### CODE REVIEW

File: roles/deprovision-buildfarm-apb/tasks/main.yml [hunk 1]
```diff
@@ -18,29 +18,46 @@
 
 - name: Delete AG-Digger Deployment Configs
   command: "oc delete dc/{{ item }} -n {{ namespace }}"
+  changed_when: false
   with_items:
   - '{{ ag_android_sdk.deployment.name }}'
   - '{{ ag_jenkins.deployment.name }}'
 
+- name: Delete AG-Jenkins Builds
+  command: "oc delete builds -n {{ namespace }} -l mobile-client-build=true"
```
Comment (maleck13) [inline roles/deprovision-buildfarm-apb/tasks/main.yml:27]: wondering if there are none found does this fails gracefully. IE just carry on?

File: roles/provision-buildfarm-apb/tasks/main.yml [hunk 2]
```diff
@@ -139,94 +145,94 @@
...
+
+  - name: "Install the Android SDK"
+    command: "oc -n {{ namespace }} rsh {{ android_sdk_podname }} androidctl sdk install"
+    when: sdk_folder.stat.exists == False
+
+  - name: Copy default config file
+    copy:
+      src: default.cfg
+      dest: /tmp/sdk.cfg
+      mode: 0777
```
Comment (maleck13) [inline roles/provision-buildfarm-apb/tasks/main.yml:171]: I know this is not part of your changes, but is there a reason for it to be this permissive?

**Comment by maleck13** [comment]
@vchepeli do you have some sample buildconfigs etc to use for verification?

**Review by maleck13 — COMMENTED** [review]
This looks good, but would like have some verification steps

**Comment by vchepeli** [comment]
@maleck13 
```yaml
---
kind: BuildConfig
apiVersion: v1
metadata:
  name: helloworld-android-pipeline1
spec:
  source:
    git:
      uri: https://github.com/odra/helloworld-android-gradle.git
      ref: jenkins-apk-path
  strategy:
    jenkinsPipelineStrategy:
      env:
        -
          name: FH_CONFIG_CONTENT
          value: empty_content
        -
          name: BUILD_CONFIG
          value: debug
      jenkinsfilePath: Jenkinsfile

```

**Review by JameelB — APPROVED** [review]

File: roles/deprovision-buildfarm-apb/tasks/main.yml (hunk 1 above)
Comment (JameelB) [reply roles/deprovision-buildfarm-apb/tasks/main.yml:27]: Tested this and it works even if there's no builds found

**Comment by witmicko** [comment]
## verified locally
👍

**Review by dimitraz — APPROVED** [review]
Verified locally :+1:

File: roles/provision-buildfarm-apb/tasks/main.yml (hunk 2 above)
Comment (ghost) [reply roles/provision-buildfarm-apb/tasks/main.yml:171]: In tmp dir it does not matter really what mode you use. @odra wdyt?

Comment (odra) [reply roles/provision-buildfarm-apb/tasks/main.yml:171]: Feel free to change it (doesn't matter as long as you can copy it into the android sdk pod), the file gets deleted together with the temporary provisioning container created by asb

**Pull request merged**